src/components/singleFile.tsx

```diff
@@ -0,0 +1,166 @@
+"use client";
+
+import React, { useState, useRef } from "react";
+import pdfPreviewImg from "../assets/images/pdf-icon.png";
+import textPreviewImg from "../assets/images/text-icon.png";
+import audioPreviewImg from "../assets/images/music-icon.png";
+import apkPreviewImg from "../assets/images/apk-icon.png";
+import zipPreviewImg from "../assets/images/zip-icon.png";
+import sqlPreviewImg from "../assets/images/sql-icon.png";
+import filePreviewImg from "../assets/images/file-icon.png";
+import { StaticImageData } from "next/image";
+
+export default function SingleFileUpload({
+  uploadedFile,
+  callback,
+  uploadBtn = "Upload",
+  progressBtn = "Uploading...",
+  pdfPreview = pdfPreviewImg,
+  textPreview = textPreviewImg,
+  audioPreview = audioPreviewImg,
+  apkPreview = apkPreviewImg,
+  zipPreview = zipPreviewImg,
+  sqlPreview = sqlPreviewImg,
+  filePreview = filePreviewImg,
+  children,
+}: any) {
+  const [isUploading, setIsUploading] = useState(false);
+  const [fileObj, setFileObj] = useState({});
+
+  const fileData = {} as {
+    previewType: string;
+    previewUrl: string | StaticImageData | ArrayBuffer | null;
+    previewName: string;
+    isDragging: boolean;
+    isLoading: boolean;
+  };
+
+  const inputRef = useRef(null);
+
+  const selectFile = () => {
+    if (isUploading) {
+      return;
+    }
+    if (inputRef.current) {
+      (inputRef.current as HTMLInputElement).click();
+    }
+  };
+
+  const handleFileChange = (event: any) => {
+    const file = event.target.files[0];
+    if (file) {
+      previewFile(file);
+    }
+  };
+
+  const previewFile = (file: any) => {
+    fileData.previewType = "image";
+    const reader = new FileReader();
+
+    reader.onload = () => {
+      if (file.type.startsWith("image/")) {
```

Comment:
Can make this function common, if its reusable

Reply:
will refactor while implement for multiple file